katdal/visdatav4.py

```diff
@@ -343,11 +334,26 @@ def __init__(self, source, ref_ant='', time_offset=0.0, applycal='',
         available_products = [product for product in CAL_PRODUCTS
                               if has_cal_product(self.sensor, attrs, product)]
         self._applycal = _selection_to_list(applycal, all=available_products)
+        if not self.source.data or not self._applycal:
+            self.correction = None
```

Comment:
`self._correction`? I also have a slight preference for `_corrections`.

Reply:
Yes, and I agree - done.